src/openforms/registrations/tasks.py

```diff
@@ -92,6 +92,16 @@ def pre_registration(submission_id: int, event: PostSubmissionEvents) -> None:
             raise exc
         return
 
+    if result.reference is None:
+        set_submission_reference(submission)
+    else:
+        submission.public_registration_reference = result.reference
+
+    if result.data is not None:
+        if not submission.registration_result:
+            submission.registration_result = {}
+        submission.registration_result.update(result.data)
```

Comment:
What would be the use case where there is already a `registration_result` set when this `pre_registration` task is being called?

Unrelated: this makes me wonder if there would be a use case where the registration result isn't a `dict` :thinking:
I guess dict is the one that makes most sense.

Reply:
I think so: if the `pre_registration` fails, then the traceback is saved in `registration_result`. So when the chain is retried, it would be already set.

Reply:
> this makes me wonder if there would be a use case where the registration result isn't a dict 🤔

Mhmm I guess that if a plugin sets it to something that is not an object it could be problematic :thinking:

Reply:
> Similarly, could Submission.save_registration_status be used?

I wasn't sure if I should use it, since the success of the `pre_registration` doesn't affect the 'status'. So I thought that it may be confusing. But it could be used like this:
```python
submission.save_registration_status(
            submission.registration_result, result.data
)
```
> shouldn't we remove the traceback key from the registration result if it succeeded?

I am also not sure if it needs to remain for possible debugging :thinking: